bigtable/client/table.cc

```diff
@@ -142,29 +149,67 @@ RowReader Table::ReadRows(RowSet row_set, Filter filter) {
 
 RowReader Table::ReadRows(RowSet row_set, std::int64_t rows_limit,
                           Filter filter) {
-  if (rows_limit <= 0) {
-    internal::RaiseInvalidArgument("rows_limit must be >0");
-  }
   return RowReader(client_, table_name(), std::move(row_set), rows_limit,
                    std::move(filter), rpc_retry_policy_->clone(),
                    rpc_backoff_policy_->clone(),
                    bigtable::internal::make_unique<
                        bigtable::internal::ReadRowsParserFactory>());
 }
 
-std::pair<bool, Row> Table::ReadRow(std::string row_key, Filter filter) {
+std::pair<bool, Row> Table::ReadRow(std::string row_key, Filter filter,
+                                    grpc::Status& status) {
   RowSet row_set(std::move(row_key));
   std::int64_t const rows_limit = 1;
   RowReader reader =
       ReadRows(std::move(row_set), rows_limit, std::move(filter));
   auto it = reader.begin();
   if (it == reader.end()) {
+    grpc::Status status = reader.finish();
```

Comment:
That should be:

```C++
    status = reader.finish();
```

Right?  And this is why we need separate unit tests btw.

Reply:
agree. I will add unit test. not sure why all test passed with this bug